product_docs/docs/pgd/5/upgrades/manual_overview.mdx

```diff
@@ -0,0 +1,224 @@
+---
+title: "Upgrading PGD clusters manually"
+---
+
+Because EDB Postgres Distributed consists of multiple software components,
+the upgrade strategy depends partially on the components that are being upgraded.
+
+In general, you can upgrade the cluster with almost zero downtime by
+using an approach called rolling upgrade. Using this approach, nodes are upgraded one by one, and
+the application connections are switched over to already upgraded nodes.
+
+You can also stop all nodes, perform the upgrade on all nodes, and
+only then restart the entire cluster. This approach is the same as with a standard PostgreSQL setup.
+This strategy of upgrading all nodes at the same time avoids running with
+mixed versions of software and therefore is the simplest. However, it incurs
+some downtime and we don't recommend it unless you can't perform the rolling upgrade
+for some reason.
+
+To upgrade an EDB Postgres Distributed cluster:
+
+1. Plan the upgrade.
+2. Prepare for the upgrade.
+3. Upgrade the server software.
+4. Check and validate the upgrade.
+
+## Upgrade planning
+
+There are broadly two ways to upgrade each node:
+
+* Upgrade nodes in place to the newer software version. See [Rolling server software upgrades](#rolling-server-software-upgrades).
+* Replace nodes with ones that have the newer version installed. See [Rolling upgrade using node join](#rolling-upgrade-using-node-join).
+
+You can use both of these approaches in a rolling manner.
+
+### Rolling upgrade considerations
+
+While the cluster is going through a rolling upgrade, mixed versions of software
+are running in the cluster. For example, suppose nodeA has PGD 3.7.16, while
+nodeB and nodeC has 4.1.0. In this state, the replication and group
+management uses the protocol and features from the oldest version (3.7.16
+in this example), so any new features provided by the newer version
+that require changes in the protocol are disabled. Once all nodes are
+upgraded to the same version, the new features are enabled.
+
+Similarly, when a cluster with WAL-decoder-enabled nodes is going through a
+rolling upgrade, WAL decoder on a higher version of PGD node produces LCRs
```

Comment:
As this is the first time we mention it on this page, how about explaining what "LCR" stands for?

Reply:
```suggestion
rolling upgrade, WAL decoder on a higher version of PGD node produces LCRs ([logical change records](..//node_management/decoding_worker/#enabling)
```